### ISSUE TITLE
BCI reproductives

### ISSUE THREAD

**Issue by Armand1** [ben18785/Selection_simulations#8]
There is an issue that has been in the back of my mind re the BCI data, namely, whether the trees that we are counting can, in fact, contribute to the next generation.  

Currently, we are modeling all the trees, regardless of how big they are. But this is not realistic.  A seed falls, grows into a shoot, it gets counted; but it cannot contribute progeny to the next census interval since it's still immature. 

One way to deal with this is to only count trees of a certain size. In fact, every tree at every census comes with a "dbh" value, "diameter at breast height". I think that one could exclude all trees with dbh<10 cm say, and that would give us reproductives.  In effect, a tree would be invisible until it reaches that size and suddenly appear in the population as reproductive individual. I just raise this now, since it may be that we need to re-shape the data and rerun the models with this constraint. I might also affect the frequencies of species that you start with in any simulations that you do. I will ask James and Chisholm what they think.

**Comment by ben18785** [comment]
Yes, interesting. I agree with your idea for rerunning the inference.

Perhaps the best way to handle this would be by individual based
simulation. Where each seed disperses from its parent, then the tree grows
(according to its particular characteristics) and competes for local space
once it has reached a given size. The idea would then be to compare
individual based simulations with the actual data and fit their parameters.

This may well have been done but could be a fun summer project for a
student!

**Comment by Armand1** [comment]
Sounds complex. I have asked Ryan what to do about this. James has flagged this issue as well.

**Comment by ben18785** [comment]
It's not complex to formulate an individual based model. My fear is that a) it may have been done already and b) we don't know enough about tree species ecologies to make such a model useful. I do think it would be cool though and a more complete way of handling this issue.

**Comment by Armand1** [comment]
I think we should set aside the notion of an "individual based" model unless you can specify, in some detail, what information we would need for it, and then I could find out whether it's obtainable. 

Continuing with our previous trajectory, I have now obtained a dataset which consists of just reproductive individuals.  Chisholm provided me with a number called "robin" which is the size at which each species becomes reproductive. This number, along with information extracted from the data itself, is also the basis for an estimate of generation time for each species, which I have now obtained (I'll explain how elsewhere https://github.com/ben18785/Selection_simulations/issues/10#issuecomment-481606541). Unfortunately, he does not have this number for all species and, when all is said and done, where we originally had 328 species in the BCI dataset, we now have 259, that is, we have lost 69.  We have lost some of these because no reproductives were observed; but most because we don't have this "robin" number.  It may be possible to fill in some of the missing robin numbers in the future with expert advice.

I attach two datasets.  The first is the BCI data themselves.  The second contains a column, for each of the 259, species with an estimate of generation time.  I think we **might** be able to make use of species-specific generation times when converting betas to selection coefficients. Alternatively, we might use the median which is **110 years**. 



[BCI_data_forben_reproductivesonly.csv.zip](https://github.com/ben18785/Selection_simulations/files/3062814/BCI_data_forben_reproductivesonly.csv.zip)
[generation_time_imputed.csv.zip](https://github.com/ben18785/Selection_simulations/files/3062819/generation_time_imputed.csv.zip)

**Comment by Armand1** [comment]
These are the data for use in the simulations with reproductives only.  The generation times are in here as well

[BCI_data_forben_reproductivesonly_with_gt.csv.zip](https://github.com/ben18785/Selection_simulations/files/3134201/BCI_data_forben_reproductivesonly_with_gt.csv.zip)
